### ISSUE TITLE
panic: runtime error: index out of range

### ISSUE THREAD

**Issue by bang88** [Originate/git-town#1095]
can not use git town sub commands after upgrade to the latest version
```sh
$ git town config

panic: runtime error: index out of range

goroutine 1 [running]:
github.com/Originate/git-town/src/git.(*ConfigMap).initialize(0xc4200442b0)
	/private/tmp/git-town-20180116-61670-tzgeis/git-town-6.0.0/src/github.com/Originate/git-town/src/git/config_map.go:82 +0x3c5
github.com/Originate/git-town/src/git.(*ConfigMap).Get(0xc4200442b0, 0x149bba4, 0x19, 0x0, 0x0)
	/private/tmp/git-town-20180116-61670-tzgeis/git-town-6.0.0/src/github.com/Originate/git-town/src/git/config_map.go:47 +0x2b
github.com/Originate/git-town/src/git.GetConfigurationValue(0x149bba4, 0x19, 0x0, 0x0)
	/private/tmp/git-town-20180116-61670-tzgeis/git-town-6.0.0/src/github.com/Originate/git-town/src/git/config.go:85 +0x41
github.com/Originate/git-town/src/git.GetMainBranch(0x0, 0x0)
	/private/tmp/git-town-20180116-61670-tzgeis/git-town-6.0.0/src/github.com/Originate/git-town/src/git/config.go:95 +0x36
github.com/Originate/git-town/src/git.GetPrintableMainBranch(0x0, 0x0)
	/private/tmp/git-town-20180116-61670-tzgeis/git-town-6.0.0/src/github.com/Originate/git-town/src/git/printable.go:13 +0x22
github.com/Originate/git-town/src/cmd.printConfig()
	/private/tmp/git-town-20180116-61670-tzgeis/git-town-6.0.0/src/github.com/Originate/git-town/src/cmd/config.go:35 +0x45
github.com/Originate/git-town/src/cmd.glob..func6(0x1724cc0, 0x174a9b0, 0x0, 0x0)
	/private/tmp/git-town-20180116-61670-tzgeis/git-town-6.0.0/src/github.com/Originate/git-town/src/cmd/config.go:24 +0x44
github.com/Originate/git-town/vendor/github.com/spf13/cobra.(*Command).execute(0x1724cc0, 0x174a9b0, 0x0, 0x0, 0x1724cc0, 0x174a9b0)
	/private/tmp/git-town-20180116-61670-tzgeis/git-town-6.0.0/src/github.com/Originate/git-town/vendor/github.com/spf13/cobra/command.go:750 +0x2c1
github.com/Originate/git-town/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x1727300, 0x138ea01, 0x14a164f, 0x25)
	/private/tmp/git-town-20180116-61670-tzgeis/git-town-6.0.0/src/github.com/Originate/git-town/vendor/github.com/spf13/cobra/command.go:831 +0x30e
github.com/Originate/git-town/vendor/github.com/spf13/cobra.(*Command).Execute(0x1727300, 0x0, 0x1)
	/private/tmp/git-town-20180116-61670-tzgeis/git-town-6.0.0/src/github.com/Originate/git-town/vendor/github.com/spf13/cobra/command.go:784 +0x2b
github.com/Originate/git-town/src/cmd.Execute()
	/private/tmp/git-town-20180116-61670-tzgeis/git-town-6.0.0/src/github.com/Originate/git-town/src/cmd/root.go:31 +0x3d
main.main()
	/private/tmp/git-town-20180116-61670-tzgeis/git-town-6.0.0/main.go:6 +0x20

```

**Comment by kevgo** [comment]
Hey @bang88, sounds like your local Git configuration contains an entry without value. Can you please run `git config -l` and post the output here? Please scrub any sensitive data from the output first.

**Comment by bang88** [comment]
@kevgo Thanks for your reply

```sh
credential.helper=osxkeychain
filter.lfs.clean=git-lfs clean -- %f
filter.lfs.smudge=git-lfs smudge -- %f
filter.lfs.process=git-lfs filter-process
filter.lfs.required=true
alias.l=log --pretty=oneline -n 20 --graph --abbrev-commit
alias.s=status -s
alias.d=!git diff-index --quiet HEAD -- || clear; git --no-pager diff --patch-with-stat
alias.di=!d() { git diff --patch-with-stat HEAD~$1; }; git diff-index --quiet HEAD -- || clear; d
alias.p=!git pull; git submodule foreach git pull origin master
alias.c=clone --recursive
alias.ca=!git add -A && git commit -av
alias.go=!f() { git checkout -b "$1" 2> /dev/null || git checkout "$1"; }; f
alias.tags=tag -l
alias.branches=branch -a
alias.remotes=remote -v
alias.aliases=config --get-regexp alias
alias.amend=commit --amend --reuse-message=HEAD
alias.credit=!f() { git commit --amend --author "$1 <$2>" -C HEAD; }; f
alias.reb=!r() { git rebase -i HEAD~$1; }; r
alias.retag=!r() { git tag -d $1 && git push origin :refs/tags/$1 && git tag $1; }; r
alias.fb=!f() { git branch -a --contains $1; }; f
alias.ft=!f() { git describe --always --contains $1; }; f
alias.fc=!f() { git log --pretty=format:'%C(yellow)%h  %Cblue%ad  %Creset%s%Cgreen  [%cn] %Cred%d' --decorate --date=short -S$1; }; f
alias.fm=!f() { git log --pretty=format:'%C(yellow)%h  %Cblue%ad  %Creset%s%Cgreen  [%cn] %Cred%d' --decorate --date=short --grep=$1; }; f
alias.dm=!git branch --merged | grep -v '\*' | xargs -n 1 git branch -d
alias.contributors=shortlog --summary --numbered
alias.mpr=!f() { 		declare currentBranch="$(git symbolic-ref --short HEAD)"; 		declare branch="${2:-$currentBranch}"; 		if [ $(printf "%s" "$1" | grep '^[0-9]\+$' > /dev/null; printf $?) -eq 0 ]; then 			git fetch origin refs/pull/$1/head:pr/$1 && 			git checkout -B $branch && 			git rebase $branch pr/$1 && 			git checkout -B $branch && 			git merge pr/$1 && 			git branch -D pr/$1 && 			git commit --amend -m "$(git log -1 --pretty=%B)

Closes #$1."; 		fi 	}; f
alias.append=town append
alias.hack=town hack
alias.kill=town kill
alias.new-pull-request=town new-pull-request
alias.prepend=town prepend
alias.prune-branches=town prune-branches
alias.rename-branch=town rename-branch
alias.repo=town repo
alias.ship=town ship
alias.sync=town sync
apply.whitespace=fix
core.excludesfile=~/.gitignore
core.attributesfile=~/.gitattributes
core.whitespace=space-before-tab,-indent-with-non-tab,trailing-space
core.trustctime=false
core.precomposeunicode=false
core.quotepath=false
color.ui=auto
color.branch.current=yellow reverse
color.branch.local=yellow
color.branch.remote=green
color.diff.meta=yellow bold
color.diff.frag=magenta bold
color.diff.old=red
color.diff.new=green
color.status.added=yellow
color.status.changed=green
color.status.untracked=cyan
diff.renames=copies
diff.indentheuristic=true
diff.bin.textconv=hexdump -v -C
help.autocorrect=1
merge.log=true
push.default=matching
push.followtags=true
url.git@github.com:.insteadof=gh:
url.git@github.com:.pushinsteadof=github:
url.git@github.com:.pushinsteadof=git://github.com/
url.git://github.com/.insteadof=github:
url.git@gist.github.com:.insteadof=gst:
url.git@gist.github.com:.pushinsteadof=gist:
url.git@gist.github.com:.pushinsteadof=git://gist.github.com/
url.git://gist.github.com/.insteadof=gist:
user.email=mymail@gmail.com
user.name=bang
user.signingkey=mykey
gpg.program=gpg
commit.gpgsign=false
filter.lfs.clean=git-lfs clean -- %f
filter.lfs.smudge=git-lfs smudge -- %f
filter.lfs.process=git-lfs filter-process
filter.lfs.required=true
```

**Comment by kevgo** [comment]
@bang88 your `mpr` alias contains newline characters. I have created #1100 with a more robust way of parsing Git configuration data that should work with your configuration. Once merged, it will be released as version 6.0.2.
